pytorch_lightning/trainer/supporters.py

```diff
@@ -371,6 +371,81 @@ def __init__(self, loaders: Any, mode: str = 'min_size'):
         if self.mode == 'max_size_cycle':
             self._wrap_loaders_max_size_cycle()
 
+        self._loaders_iter_state_dict = None
+        self._iterator = None  # assigned in __iter__
+
+    def state_dict(self, num_batches_processed: int) -> Dict:
+        """
+        The state dict includes all states from wrapped dataloaders and their samplers through the
+        ``CaptureIterableDataset`` and fast-forward samplers.
+
+        Args:
+            num_batches_processed: The number of batches processed so far, needed because the individual dataloaders
+                may have already prefetched more batches by the time a state dict is requested.
+        """
+        if not _fault_tolerant_enabled():
+            return DataLoaderDict()
+
+        def state_dict_fn(dataloader: DataLoader, iterator: Optional[Iterator]) -> Dict:
```

Comment:
just curious, why it is inside function and function on its own?

Reply:
Moved to its own function.